components/question/Details/index.jsx

```diff
@@ -21,11 +35,71 @@ const QuestionDetails = ({ setBackgroundImagePath }) => {
     }
   }, [survey, setBackgroundImagePath]);
 
+  useEffect(() => {
+    if (isSurveySubmitted) {
+      setBackgroundImagePath('');
+    }
+  }, [isSurveySubmitted, setBackgroundImagePath]);
+
   if (!survey) {
     return null;
   }
 
-  return <div data-test-id={questionDetailsTestIds.base}>TODO: Question rendering goes here</div>;
+  const questions = getQuestionsFromSurvey(survey);
+  const surveyQuestions = questions.filter(question => question.type !== 'intro' && question.type !== 'outro');
+  const currentQuestion = surveyQuestions[currentQuestionIndex];
+  const currentQuestionSerial = `${currentQuestionIndex + 1} / ${surveyQuestions.length}`;
+  const isLastQuestion = currentQuestionIndex + 1 === surveyQuestions.length;
+  const QuestionComponent = getQuestionComponentFromQuestionType(currentQuestion.type);
+
+  if (isSurveySubmitted) {
+    return <div>TODO: Show a thank you message</div>;
+  }
+
+  const renderQuestionComponent = () =>
+    QuestionComponent && (
+      <QuestionComponent {...currentQuestion} survey={survey} setSurveyQuestionsWithAnswers={setSurveyQuestionsWithAnswers} />
+    );
+
+  const goToNextQuestion = () =>
+    isLastQuestion ? (
+      <SurveySubmit
+        surveyId={surveyId}
+        surveyQuestionsWithAnswers={surveyQuestionsWithAnswers}
+        setIsSurveySubmitted={setIsSurveySubmitted}
+      />
+    ) : (
+      <div className="w-14 h-14 fixed bottom-0 right-0 mb-8 mr-8">
+        <Image
+          className="cursor-pointer"
+          src={angleRightIcon}
+          alt="next question"
+          data-test-id={questionDetailsTestIds.angleRightIcon}
+          onClick={() => setCurrentQuestionIndex(currentIndex => currentIndex + 1)}
+        />
+      </div>
```

Comment:
Should we use guard clause to make it clearer?
```suggestion
    if (isLastQuestion) {
       return <SurveySubmit
        surveyId={surveyId}
        surveyQuestionsWithAnswers={surveyQuestionsWithAnswers}
        setIsSurveySubmitted={setIsSurveySubmitted}
      />
    }
    
    return (
      <div className="w-14 h-14 fixed bottom-0 right-0 mb-8 mr-8">
        <Image
          className="cursor-pointer"
          src={angleRightIcon}
          alt="next question"
          data-test-id={questionDetailsTestIds.angleRightIcon}
          onClick={() => setCurrentQuestionIndex(currentIndex => currentIndex + 1)}
        />
      </div>
      )
```

Reply:
sure, that's a nice suggestions, implemented in https://github.com/tyrro/react-survey/pull/62/commits/3544da1fee01753798712777f1ecdf2fea723eb2, Thanks!